server.js

```diff
@@ -4549,28 +4549,31 @@ cache(function(data, match, sendBadge, request) {
       return;
     }
     try {
-      var status = data[0].status;
-      switch(status) {
-      case 'success':
-      case 'fixed':
-        badgeData.colorscheme = 'brightgreen';
-        badgeData.text[1] = 'passing';
-        break;
+      try {
+        var status = data[0].status;
+        switch(status) {
+        case 'success':
+        case 'fixed':
+          badgeData.colorscheme = 'brightgreen';
+          badgeData.text[1] = 'passing';
+          break;
 
-      case 'failed':
-        badgeData.colorscheme = 'red';
-        badgeData.text[1] = 'failed';
-        break;
+        case 'failed':
+          badgeData.colorscheme = 'red';
+          badgeData.text[1] = 'failed';
+          break;
 
-      case 'no_tests':
-      case 'scheduled':
-      case 'not_run':
-        badgeData.colorscheme = 'yellow';
-      default:
-        badgeData.text[1] = status.replace('_', ' ');
+        case 'no_tests':
+        case 'scheduled':
+        case 'not_run':
+          badgeData.colorscheme = 'yellow';
+        default:
+          badgeData.text[1] = status.replace('_', ' ');
         break;
+        }
+      } catch(e) {
+        badgeData.text[1] = data.message;
```

Comment:
its more for if there is an error and `data.message` is present then show the user the message so they might be able to work out for themselves how to fix the image url.

In the following cases:

`Permission denied`:
* the user has supplied a key with insufficient/incorrect privileges.

`Project not found`:
* the project does not exist on Circle CI (maybe mistyped)
* no key was supplied when it was necessary to do so (private)

There may be more, but i haven't seen any others yet.

If you prefer to keep it the way it is now i understand though as this could also lead to more confusion than just showing a standard message.